### PR TITLE
Add addon config provider

### PR DESCRIPTION
**Pull request by maciaszczykm**
**What this PR does / why we need it**: Currently get and list methods for addon config return 502 error as current addon provider requires DC as a part of the request. This is not required for addon configs so I have added a new provider specific for them.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:

**Special notes for your reviewer**:

**Documentation**:

**Does this PR introduce a user-facing change?**:

```release-note
Fix addon config get and list endpoints.
```

### CODE REVIEW

**Comment by maciaszczykm** [comment]
PTAL @zreigz

**Review by zreigz — APPROVED** [review]
/lgtm
/approve

File: api/pkg/provider/kubernetes/addonconfig.go [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+func NewAddonConfigProvider(client kubermaticclientset.Interface, addonConfigLister kubermaticv1lister.AddonConfigLister) *AddonConfigProvider {
+	return &AddonConfigProvider{
+		client:            client,
+		addonConfigLister: addonConfigLister,
+	}
+}
+
+// Get addon configuration
+func (p *AddonConfigProvider) Get(addonName string) (*kubermaticv1.AddonConfig, error) {
+	return p.client.KubermaticV1().AddonConfigs().Get(addonName, metav1.GetOptions{})
```
Comment (alvaroaleman) [inline api/pkg/provider/kubernetes/addonconfig.go:27]: Err no, we must use the impersonation client, this allows any user to get the addon config of any cluster

**Review by alvaroaleman — APPROVED** [review]
/lgtm
/approve

**Pull request merged**